src/rules/no-extraneous-dependencies.js

```diff
@@ -1,23 +1,40 @@
 import path from 'path'
+import fs from 'fs'
 import readPkgUp from 'read-pkg-up'
 import minimatch from 'minimatch'
 import importType from '../core/importType'
 import isStaticRequire from '../core/staticRequire'
 
-function getDependencies(context) {
+function getDependencies(context, packageDir) {
   try {
-    const pkg = readPkgUp.sync({cwd: context.getFilename(), normalize: false})
-    if (!pkg || !pkg.pkg) {
+    const packageContent = packageDir
+      ? JSON.parse(fs.readFileSync(path.join(packageDir, 'package.json'), 'utf8'))
+      : readPkgUp.sync({cwd: context.getFilename(), normalize: false}).pkg
+
+    if (!packageContent) {
       return null
     }
-    const packageContent = pkg.pkg
+
     return {
       dependencies: packageContent.dependencies || {},
       devDependencies: packageContent.devDependencies || {},
       optionalDependencies: packageContent.optionalDependencies || {},
       peerDependencies: packageContent.peerDependencies || {},
     }
   } catch (e) {
+    if (packageDir && e.code === 'ENOENT') {
+      context.report({
+        message: 'The package.json file could not be found.',
+        loc: { line: 0, column: 0 },
+      })
+    }
+    if (packageDir && e instanceof SyntaxError) {
```

Comment:
Ready, `read-pkg-up` outputs a different kind of error for json syntax errors https://github.com/sindresorhus/parse-json/blob/master/index.js#L5